internal/connect/command_offset_update.go

```diff
@@ -0,0 +1,142 @@
+package connect
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	connectv1 "github.com/confluentinc/ccloud-sdk-go-v2/connect/v1"
+
+	pcmd "github.com/confluentinc/cli/v3/pkg/cmd"
+	"github.com/confluentinc/cli/v3/pkg/errors"
+	"github.com/confluentinc/cli/v3/pkg/examples"
+	"github.com/confluentinc/cli/v3/pkg/kafka"
+	"github.com/confluentinc/cli/v3/pkg/output"
+	"github.com/confluentinc/cli/v3/pkg/retry"
+)
+
+func (c *offsetCommand) newUpdateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:               "update <id>",
+		Short:             "Update a connector's offsets.",
+		Args:              cobra.ExactArgs(1),
+		RunE:              c.update,
+		ValidArgsFunction: pcmd.NewValidArgsFunction(c.validArgs),
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: "Update offsets for a connector in the current or specified Kafka cluster context.",
+				Code: "confluent connect offset update lcc-123456 --config-file config.json",
+			},
+			examples.Example{
+				Code: "confluent connect offset update lcc-123456 --config-file config.json --cluster lkc-123456",
+			},
+		),
+	}
+
+	cmd.Flags().String("config-file", "", "JSON file containing connector offsets to set to.")
+	pcmd.AddClusterFlag(cmd, c.AuthenticatedCLICommand)
+	pcmd.AddContextFlag(cmd, c.CLICommand)
+	pcmd.AddEnvironmentFlag(cmd, c.AuthenticatedCLICommand)
+	pcmd.AddOutputFlag(cmd)
+
+	cobra.CheckErr(cmd.MarkFlagFilename("config-file", "json"))
+	cobra.CheckErr(cmd.MarkFlagRequired("config-file"))
+
+	return cmd
+}
+
+func (c *offsetCommand) update(cmd *cobra.Command, args []string) error {
+	kafkaCluster, err := kafka.GetClusterForCommand(c.V2Client, c.Context)
+	if err != nil {
+		return err
+	}
+
+	environmentId, err := c.Context.EnvironmentId()
+	if err != nil {
+		return err
+	}
+
+	connector, err := c.V2Client.GetConnectorExpansionById(args[0], environmentId, kafkaCluster.ID)
+	if err != nil {
+		return err
+	}
+
+	connectorName := connector.Info.GetName()
+	configFile, err := cmd.Flags().GetString("config-file")
+	if err != nil {
+		return err
+	}
+
+	request, err := c.getAlterOffsetRequestBody(configFile)
+	if err != nil {
+		return err
+	}
+
+	_, err = c.V2Client.AlterConnectorOffsets(connectorName, environmentId, kafkaCluster.ID, *request)
+	if err != nil {
+		return err
+	}
+
+	var offsetStatus connectv1.ConnectV1AlterOffsetStatus
+
+	err = retry.Retry(time.Second, 30*time.Second, func() error {
```

Comment:
If the time expires and the phase is still pending, I think we should throw an error to the user and ask them to use the status api to check for the status manually. 

Along those lines, should we make this timeout configurable with 0 returning immediately.

Reply:
Also please refer to the comments I made on the delete CLI and incorporate the changes here as well.